packages/next/src/server-build.ts

```diff
@@ -1090,6 +1089,36 @@ export async function serverBuild({
     )
   );
 
+  let pagesPlaceholderRscEntries;
+
+  if (appDir) {
+    // since we attempt to rewrite all paths to an .rsc variant,
+    // we need to create dummy rsc outputs for all pages entries
+    // this is so that an RSC request to a `pages` entry will match
+    // rather than falling back to a catchall `app` entry
+    // on the nextjs side, invalid RSC response payloads will correctly trigger an mpa navigation
+    const pagesManifest = path.join(
+      entryPath,
+      outputDirectory,
+      `server/pages-manifest.json`
+    );
+
+    const pagesData = await fs.readJSON(pagesManifest);
+    const pagesEntries = Object.keys(pagesData);
+
+    if (pagesEntries.length > 0) {
+      const dummyFilePath = path.join('./', 'dummyFile');
```

Comment:
Is there a difference between `path.join('./', 'dummyFile')` and the shorter `./dummyFile`?

Was the latter not working on windows?

Reply:
I lifted this from https://github.com/vercel/vercel/blob/main/packages/next/src/server-build.ts#L1785 assuming there was a OS-specific reason, but I think I'm going to update this to write to `tmpdir()` since that feels more appropriate (unless there's an easier way to do this without writing anything to disk? 🤔 )